packages/server-core/src/moderation/moderation/moderation.resolvers.ts

```diff
@@ -35,10 +36,30 @@ import {
 import { fromDateTimeSql, getDateTimeSql } from '@ir-engine/common/src/utils/datetime-sql'
 import type { HookContext } from '@ir-engine/server-core/declarations'
 
+const resolveUserEmail = async (userId: UserID | undefined, context: HookContext) => {
+  if (!userId) return undefined
+
+  const identityProvider = await context.app.service(identityProviderPath).find({
+    isInternal: true,
+    query: {
+      userId: userId,
+      $limit: 1
+    }
+  })
+  return identityProvider?.data[0]?.email || undefined
+}
+
 export const moderationResolver = resolve<ModerationType, HookContext>({
   createdAt: virtual(async (moderation) => fromDateTimeSql(moderation.createdAt)),
-  updatedAt: virtual(async (moderation) => fromDateTimeSql(moderation.updatedAt))
+  updatedAt: virtual(async (moderation) => fromDateTimeSql(moderation.updatedAt)),
+  reportedUserEmail: virtual(async (moderation: ModerationType, context: HookContext) => {
```

Comment:
These should be in external resolver only instead of this and should only work for find operation